Source/Core/VideoCommon/RenderBase.cpp

```diff
@@ -676,7 +676,16 @@ void Renderer::Swap(u32 xfbAddr, u32 fbWidth, u32 fbStride, u32 fbHeight, const
       m_last_xfb_ticks = ticks;
 
       auto xfb_rect = texture_config.GetRect();
-      xfb_rect.right -= EFBToScaledX(fbStride - fbWidth);
+
+      // It's possible that the returned XFB texture is native resolution
+      // even when we're rendering at higher than native resolution
+      // if the XFB was was loaded entirely from console memory.
+      // If so, adjust the rectangle by native resolution instead of scaled resolution.
+      const u32 stride_width_difference = fbStride - fbWidth;
+      if (texture_config.width == fbStride)
```

Comment:
Would it be better to compare against the cache entry's native width (texture_config.width == xfb_entry->native_width), instead of the stride here? IIRC it probably won't make a difference, as the cache won't be hit if the VI stride differs from the copy stride, but it could be clearer.

Reply:
Sure, that seems fine to me.